runtime/lua/vim/lsp/util.lua

```diff
@@ -1484,6 +1502,48 @@ function M.stylize_markdown(bufnr, contents, opts)
   return stripped
 end
 
+--- @class lsp.util.NormalizeMarkdownOptions
+--- @field width integer Thematic breaks are expanded to this size. Defaults to 80.
+
+--- Normalizes Markdown input to a canonical form.
+---
+--- The returned Markdown adheres to the GitHub Flavored Markdown (GFM)
+--- specification.
+---
+--- The following transformations are made:
+---
+---   1. Empty lines at the beginning or end of the content are removed
+---   2. Carriage returns ('\r') are removed
+---   3. Successive empty lines are collapsed into a single empty line
+---   4. Thematic breaks are expanded to the given width
+---
+---@param contents string[]
+---@param opts? lsp.util.NormalizeMarkdownOptions
+---@return string[] table of lines containing normalized Markdown
+---@see https://github.github.com/gfm
+function M.normalize_markdown(contents, opts)
```

Comment:
Sorry to do this to you, but I think *for now* this function should be marked `@private` and prefixed with a `_` (to indicate that it’s an internal-use only function).

As we’ve discussed, this function will not stay here, it will be moved out of lsp.util at some point (hopefully soon).  In the meantime, we *do not* want people depending on it, and if they do it should be very clear that it’s unstable.  We *certainly* don’t want it to make it into a release while it is public. The `@private` annotation and `_` prefix buy us some time.

Reply:
I understand, but I also don't like us deprecating `stylize_markdown` without providing a public alternative.

Reply:
We can hold off on deprecating `stylize_markdown` for now.